From f80d53ec462156d238ecc309b6a5897f09ad354f Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Tue, 28 Mar 2023 19:23:10 +0200
Subject: [PATCH 304/954] fix(st-ddr): remove typedefs

Remove typedefs in STM32MP2 DDR PHY driver.
While at it, use capital letter for enum entries.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 51b2bec01e3983892a9c8d18d2265b43bf6ec864 fix(st-ddr): remove typedefs]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: I7e10560e33cb5ee938ab2b5b9e9450dac3d98e1b
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/300203
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../firmware/include/mnpmusrammsgblock_ddr3.h |  6 +--
 .../firmware/include/mnpmusrammsgblock_ddr4.h |  6 +--
 .../include/mnpmusrammsgblock_lpddr4.h        |  6 +--
 .../ddr/phy/phyinit/include/ddrphy_phyinit.h  | 16 +++----
 .../phyinit/include/ddrphy_phyinit_struct.h   | 18 +++----
 .../include/ddrphy_phyinit_usercustom.h       | 47 ++++++++++---------
 .../src/ddrphy_phyinit_c_initphyconfig.c      | 14 +++---
 .../phyinit/src/ddrphy_phyinit_f_loaddmem.c   |  4 +-
 .../phy/phyinit/src/ddrphy_phyinit_globals.c  | 16 +++----
 .../phyinit/src/ddrphy_phyinit_initstruct.c   |  4 +-
 .../phyinit/src/ddrphy_phyinit_load_sr_fw.c   |  4 +-
 .../phyinit/src/ddrphy_phyinit_mapdrvstren.c  | 30 +++++-------
 .../phyinit/src/ddrphy_phyinit_reginterface.c | 32 ++++++-------
 .../src/ddrphy_phyinit_restore_sequence.c     |  4 +-
 .../phy/phyinit/src/ddrphy_phyinit_sequence.c |  6 +--
 .../src/ddrphy_phyinit_storeincvfile.c        |  6 +--
 .../ddrphy_phyinit_usercustom_saveretregs.c   |  2 +-
 17 files changed, 108 insertions(+), 113 deletions(-)

diff --git a/drivers/st/ddr/phy/firmware/include/mnpmusrammsgblock_ddr3.h b/drivers/st/ddr/phy/firmware/include/mnpmusrammsgblock_ddr3.h
index cfcac3626e..d171b8bc48 100644
--- a/drivers/st/ddr/phy/firmware/include/mnpmusrammsgblock_ddr3.h
+++ b/drivers/st/ddr/phy/firmware/include/mnpmusrammsgblock_ddr3.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -13,7 +13,7 @@
  * Please refer to the Training Firmware App Note for futher information about
  * the usage for Message Block.
  */
-typedef struct _pmu_smb_ddr_1d_t {
+struct pmu_smb_ddr_1d {
 	uint8_t reserved00;		/*
 					 * Byte offset 0x00, CSR Addr 0x54000, Direction=In
 					 * reserved00[0:4] RFU, must be zero
@@ -994,6 +994,6 @@ typedef struct _pmu_smb_ddr_1d_t {
 	uint8_t reserveda1;		/* Byte offset 0xa1, CSR Addr 0x54050, Direction=N/A */
 	uint8_t reserveda2;		/* Byte offset 0xa2, CSR Addr 0x54051, Direction=N/A */
 	uint8_t reserveda3;		/* Byte offset 0xa3, CSR Addr 0x54051, Direction=N/A */
-} __attribute__ ((packed)) __attribute__ ((aligned(2))) pmu_smb_ddr_1d_t;
+} __attribute__ ((packed)) __attribute__ ((aligned(2)));
 
 #endif /* MNPMUSRAMMSGBLOCK_DDR3_H */
diff --git a/drivers/st/ddr/phy/firmware/include/mnpmusrammsgblock_ddr4.h b/drivers/st/ddr/phy/firmware/include/mnpmusrammsgblock_ddr4.h
index 4a6391b564..c5d076f05a 100644
--- a/drivers/st/ddr/phy/firmware/include/mnpmusrammsgblock_ddr4.h
+++ b/drivers/st/ddr/phy/firmware/include/mnpmusrammsgblock_ddr4.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -12,7 +12,7 @@
  * Please refer to the Training Firmware App Note for futher information about
  * the usage for Message Block.
  */
-typedef struct _pmu_smb_ddr_1d_t {
+struct pmu_smb_ddr_1d {
 	uint8_t reserved00;		/*
 					 * Byte offset 0x00, CSR Addr 0x54000, Direction=In
 					 * reserved00[0:4] RFU, must be zero
@@ -2269,6 +2269,6 @@ typedef struct _pmu_smb_ddr_1d_t {
 					 *   0x1 = Do not reset DRAM during devinit
 					 *   0x0 = Reset DRAM during devinit
 					 */
-} __attribute__ ((packed)) __attribute__ ((aligned(2))) pmu_smb_ddr_1d_t;
+} __attribute__ ((packed)) __attribute__ ((aligned(2)));
 
 #endif /* MNPMUSRAMMSGBLOCK_DDR4_H */
diff --git a/drivers/st/ddr/phy/firmware/include/mnpmusrammsgblock_lpddr4.h b/drivers/st/ddr/phy/firmware/include/mnpmusrammsgblock_lpddr4.h
index 1d46a04393..885bef7a6c 100644
--- a/drivers/st/ddr/phy/firmware/include/mnpmusrammsgblock_lpddr4.h
+++ b/drivers/st/ddr/phy/firmware/include/mnpmusrammsgblock_lpddr4.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -12,7 +12,7 @@
  * Please refer to the Training Firmware App Note for futher information about
  * the usage for Message Block.
  */
-typedef struct _pmu_smb_ddr_1d_t {
+struct pmu_smb_ddr_1d {
 	uint8_t reserved00;		/*
 					 * Byte offset 0x00, CSR Addr 0x54000, Direction=In
 					 * reserved00[0:4] RFU, must be zero
@@ -962,6 +962,6 @@ typedef struct _pmu_smb_ddr_1d_t {
 	uint8_t reserved87;		/* Byte offset 0x87, CSR Addr 0x54043, Direction=N/A */
 	uint8_t reserved88;		/* Byte offset 0x88, CSR Addr 0x54044, Direction=N/A */
 	uint8_t reserved89;		/* Byte offset 0x89, CSR Addr 0x54044, Direction=N/A */
-} __attribute__ ((packed)) __attribute__ ((aligned(2))) pmu_smb_ddr_1d_t;
+} __attribute__ ((packed)) __attribute__ ((aligned(2)));
 
 #endif /* MNPMUSRAMMSGBLOCK_LPDDR4_H */
diff --git a/drivers/st/ddr/phy/phyinit/include/ddrphy_phyinit.h b/drivers/st/ddr/phy/phyinit/include/ddrphy_phyinit.h
index 97603e1bbc..5f0cc0ebce 100644
--- a/drivers/st/ddr/phy/phyinit/include/ddrphy_phyinit.h
+++ b/drivers/st/ddr/phy/phyinit/include/ddrphy_phyinit.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -12,15 +12,15 @@
 #define DDRPHY_PHYINIT_RID 202006
 
 /* Global Structures : instantiated in ddrphy_globals.c */
-extern runtime_config_t runtimeconfig;
+extern struct runtime_config runtimeconfig;
 
-extern user_input_basic_t userinputbasic;
-extern user_input_advanced_t userinputadvanced;
-extern user_input_mode_register_t userinputmoderegister;
-extern user_input_swizzle_t userinputswizzle;
+extern struct user_input_basic userinputbasic;
+extern struct user_input_advanced userinputadvanced;
+extern struct user_input_mode_register userinputmoderegister;
+extern struct user_input_swizzle userinputswizzle;
 
-extern pmu_smb_ddr_1d_t mb_ddr_1d[NB_PS];
-extern pmu_smb_ddr_1d_t shdw_ddr_1d[NB_PS];
+extern struct pmu_smb_ddr_1d mb_ddr_1d[NB_PS];
+extern struct pmu_smb_ddr_1d shdw_ddr_1d[NB_PS];
 
 /* Function definitions */
 int ddrphy_phyinit_softsetmb(int ps, char *field, int value);
diff --git a/drivers/st/ddr/phy/phyinit/include/ddrphy_phyinit_struct.h b/drivers/st/ddr/phy/phyinit/include/ddrphy_phyinit_struct.h
index f381a269e1..44bc70bd78 100644
--- a/drivers/st/ddr/phy/phyinit/include/ddrphy_phyinit_struct.h
+++ b/drivers/st/ddr/phy/phyinit/include/ddrphy_phyinit_struct.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -34,7 +34,7 @@ enum dimmtypes {
  * The following basic data structure must be set and completed correctly so
  * that the PhyInit software package can accurate program PHY registers.
  */
-typedef struct user_input_basic {
+struct user_input_basic {
 	int dramtype; /* DRAM Module Type */
 
 	/*
@@ -234,7 +234,7 @@ typedef struct user_input_basic {
 	 * - dfimode register is programed based on other parameters
 	 * - This exists for backward compatibility and no longer used.
 	 */
-} user_input_basic_t;
+};
 
 
 /* Structure for advanced (optional) user inputs
@@ -242,7 +242,7 @@ typedef struct user_input_basic {
  * If user does not enter a value for these parameters, a default recommended or
  * default value will be used
  */
-typedef struct user_input_advanced {
+struct user_input_advanced {
 	int lp4rxpreamblemode[NB_PS]; /* Selects between DRAM static read vs toggle read preamble */
 
 	/*
@@ -1028,7 +1028,7 @@ typedef struct user_input_advanced {
 					 * sequencectrl[13] = RFU, must be zero
 					 * sequencectrl[15-14] = RFU, must be zero
 					 */
-} user_input_advanced_t;
+};
 
 /*
  * Structure for mode register (mandatory) user inputs
@@ -1040,7 +1040,7 @@ typedef struct user_input_advanced {
  * - DDR4: mr0..6 are used (16-bits values)
  * - LPDDR4: mr1..4 and mr11..22 are used (8-bits values)
  */
-typedef struct user_input_mode_register {
+struct user_input_mode_register {
 	int mr0[NB_PS];
 	int mr1[NB_PS];
 	int mr2[NB_PS];
@@ -1053,7 +1053,7 @@ typedef struct user_input_mode_register {
 	int mr13[NB_PS];
 	int mr14[NB_PS];
 	int mr22[NB_PS];
-} user_input_mode_register_t;
+};
 
 /*
  * Structure for swizzle (mandatory) user inputs
@@ -1075,9 +1075,9 @@ typedef struct user_input_mode_register {
 #define NB_MAPCAATODFI_SWIZZLE		6
 #define NB_MAPCABTODFI_SWIZZLE		6
 #define NB_SWIZZLE	44
-typedef struct user_input_swizzle {
+struct user_input_swizzle {
 	int swizzle[NB_SWIZZLE];
-} user_input_swizzle_t;
+};
 
 #endif /* DDRPHY_PHYINIT_STRUCT_H */
 
diff --git a/drivers/st/ddr/phy/phyinit/include/ddrphy_phyinit_usercustom.h b/drivers/st/ddr/phy/phyinit/include/ddrphy_phyinit_usercustom.h
index b4c659e202..a70233d94d 100644
--- a/drivers/st/ddr/phy/phyinit/include/ddrphy_phyinit_usercustom.h
+++ b/drivers/st/ddr/phy/phyinit/include/ddrphy_phyinit_usercustom.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -98,41 +98,44 @@
  */
 
 /* A structure used to SRAM memory address space. */
-typedef enum { return_offset, return_lastaddr } return_offset_lastaddr_t;
+enum return_offset_lastaddr {
+	RETURN_OFFSET,
+	RETURN_LASTADDR
+};
 
 /* A structure to store the sequence function runtime input variables. */
-typedef struct runtime_config {
+struct runtime_config {
 	bool	skip_train;	/* skip_train input parameter */
 	bool	reten;		/*
 				 * Retention Enable input parameter, instructs phyinit to issue
 				 * register reads during initialization to retention registers.
 				 */
-} runtime_config_t;
+};
 
 /* Enumeration of instructions for PhyInit Register Interface */
-typedef enum {
-	starttrack,	/* Start register tracking */
-	stoptrack,	/* Stop register tracking */
-	saveregs,	/* Save(read) tracked register values */
-	restoreregs,	/* Restore (write) saved register values */
-	dumpregs,	/* Write register address,value pairs to file */
-	importregs	/* Import register address,value pairs to file */
-} reginstr;
+enum reginstr {
+	STARTTRACK,	/* Start register tracking */
+	STOPTRACK,	/* Stop register tracking */
+	SAVEREGS,	/* Save(read) tracked register values */
+	RESTOREREGS,	/* Restore (write) saved register values */
+	DUMPREGS,	/* Write register address,value pairs to file */
+	IMPORTREGS	/* Import register address,value pairs to file */
+} ;
 
 /* Data structure to store register address, value pairs */
-typedef struct reg_addr_val {
+struct reg_addr_val {
 	uint32_t	address;	/* Register address */
 	uint16_t	value;		/* Register value */
-} reg_addr_val_t;
+};
 
 /* TargetCSR Target CSR for the impedance value for ddrphy_phyinit_mapdrvstren() */
 enum drvtype {
-	drvstrenfsdqp,
-	drvstrenfsdqn,
-	odtstrenp,
-	odtstrenn,
-	adrvstrenp,
-	adrvstrenn
+	DRVSTRENFSDQP,
+	DRVSTRENFSDQN,
+	ODTSTRENP,
+	ODTSTRENN,
+	ADRVSTRENP,
+	ADRVSTRENN
 };
 
 /*
@@ -162,12 +165,12 @@ int ddrphy_phyinit_mapdrvstren(int drvstren_ohm, enum drvtype targetcsr);
 void ddrphy_phyinit_storemsgblk(void *msgblkptr, int sizeofmsgblk, int mem[]);
 int ddrphy_phyinit_calcmb(void);
 int ddrphy_phyinit_storeincvfile(char *incv_file_name, int mem[],
-				 return_offset_lastaddr_t return_type);
+				 enum return_offset_lastaddr return_type);
 void ddrphy_phyinit_writeoutmem(uint32_t *mem, int mem_offset, int mem_size);
 void ddrphy_phyinit_writeoutmsgblk(uint16_t *mem, int mem_offset, int mem_size);
 int ddrphy_phyinit_isdbytedisabled(int dbytenumber);
 int ddrphy_phyinit_trackreg(uint32_t adr);
-int ddrphy_phyinit_reginterface(reginstr myreginstr, uint32_t adr, uint16_t dat);
+int ddrphy_phyinit_reginterface(enum reginstr myreginstr, uint32_t adr, uint16_t dat);
 
 extern void ddrphy_phyinit_usercustom_a_bringuppower(void);
 extern void ddrphy_phyinit_usercustom_b_startclockresetphy(void);
diff --git a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_c_initphyconfig.c b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_c_initphyconfig.c
index 9e87a49d8f..277b1ac457 100644
--- a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_c_initphyconfig.c
+++ b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_c_initphyconfig.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -449,14 +449,14 @@ static int txodtdrvstren_program(void)
 
 		odtstrenp_state = ddrphy_phyinit_mapdrvstren(
 							(int)userinputadvanced.odtimpedance[pstate],
-							odtstrenp);
+							ODTSTRENP);
 		if (odtstrenp_state < 0) {
 			return odtstrenp_state;
 		}
 
 		odtstrenn_state = ddrphy_phyinit_mapdrvstren(
 							(int)userinputadvanced.odtimpedance[pstate],
-							odtstrenn);
+							ODTSTRENN);
 		if (odtstrenn_state < 0) {
 			return odtstrenn_state;
 		}
@@ -510,14 +510,14 @@ static int tximpedancectrl1_program(void)
 
 		drvstrenfsdqp_state = ddrphy_phyinit_mapdrvstren(
 							(int)userinputadvanced.tximpedance[pstate],
-							drvstrenfsdqp);
+							DRVSTRENFSDQP);
 		if (drvstrenfsdqp_state < 0) {
 			return drvstrenfsdqp_state;
 		}
 
 		drvstrenfsdqn_state = ddrphy_phyinit_mapdrvstren(
 							(int)userinputadvanced.tximpedance[pstate],
-							drvstrenfsdqn);
+							DRVSTRENFSDQN);
 		if (drvstrenfsdqn_state < 0) {
 			return drvstrenfsdqn_state;
 		}
@@ -564,13 +564,13 @@ static int atximpedance_program(void)
 	uint16_t atximpedance;
 
 	adrvstrenp_state = ddrphy_phyinit_mapdrvstren((int)userinputadvanced.atximpedance,
-						      adrvstrenp);
+						      ADRVSTRENP);
 	if (adrvstrenp_state < 0) {
 		return adrvstrenp_state;
 	}
 
 	adrvstrenn_state = ddrphy_phyinit_mapdrvstren((int)userinputadvanced.atximpedance,
-						      adrvstrenn);
+						      ADRVSTRENN);
 	if (adrvstrenn_state < 0) {
 		return adrvstrenn_state;
 	}
diff --git a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_f_loaddmem.c b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_f_loaddmem.c
index 2709c9d8dd..5dd486ae86 100644
--- a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_f_loaddmem.c
+++ b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_f_loaddmem.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -27,7 +27,7 @@
  */
 int ddrphy_phyinit_f_loaddmem(int pstate)
 {
-	pmu_smb_ddr_1d_t *msgblkptr;
+	struct pmu_smb_ddr_1d *msgblkptr;
 	int sizeofmsgblk;
 	uint16_t *ptr16;
 	uint32_t *ptr32;
diff --git a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_globals.c b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_globals.c
index 43ad7d6f49..e03c94047f 100644
--- a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_globals.c
+++ b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_globals.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -9,16 +9,16 @@
 #include <ddrphy_phyinit.h>
 
 /* Global struct defines */
-runtime_config_t runtimeconfig;
-user_input_basic_t userinputbasic;
-user_input_advanced_t userinputadvanced;
-user_input_mode_register_t userinputmoderegister;
-user_input_swizzle_t userinputswizzle;
+struct runtime_config runtimeconfig;
+struct user_input_basic userinputbasic;
+struct user_input_advanced userinputadvanced;
+struct user_input_mode_register userinputmoderegister;
+struct user_input_swizzle userinputswizzle;
 
 /* Firmware 1D Message Block structures */
-pmu_smb_ddr_1d_t mb_ddr_1d[NB_PS];
+struct pmu_smb_ddr_1d mb_ddr_1d[NB_PS];
 /* Shadow of 1D message block. Used by PhyInit to track user changes to the data structure */
-pmu_smb_ddr_1d_t shdw_ddr_1d[NB_PS];
+struct pmu_smb_ddr_1d shdw_ddr_1d[NB_PS];
 
 /*
  * Represent the value stored in Step C into the register with the same name.
diff --git a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_initstruct.c b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_initstruct.c
index e4cef3a928..e71dd9666e 100644
--- a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_initstruct.c
+++ b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_initstruct.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -447,7 +447,7 @@ void ddrphy_phyinit_initstruct(void)
 		mb_ddr_1d[myps].share2dvrefresult = share2dvrefresult;
 
 		/* Zero out struct contents */
-		memset((void *) &shdw_ddr_1d[myps], 0, sizeof(pmu_smb_ddr_1d_t));
+		memset((void *) &shdw_ddr_1d[myps], 0, sizeof(struct pmu_smb_ddr_1d));
 	} /* myps */
 
 	VERBOSE("%s End\n", __func__);
diff --git a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_load_sr_fw.c b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_load_sr_fw.c
index 7cf8a23d12..d80d317649 100644
--- a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_load_sr_fw.c
+++ b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_load_sr_fw.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -21,7 +21,7 @@ void ddrphy_phyinit_load_sr_fw(void)
 	int i;
 	int imem[SR_IMEM_SIZE];
 	int imem_offset = 0;
-	return_offset_lastaddr_t return_type = return_offset;
+	enum return_offset_lastaddr return_type = RETURN_OFFSET;
 
 	VERBOSE("Start of loading Universal Resident Retention IMEM\n");
 
diff --git a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_mapdrvstren.c b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_mapdrvstren.c
index 4f0074bb5d..41d11e7b14 100644
--- a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_mapdrvstren.c
+++ b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_mapdrvstren.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -22,12 +22,12 @@
  *
  * @param[in] targetcsr Target CSR for the impedance value. on of following
  * enum drvtype:
- *   - drvstrenfsdqp
- *   - drvstrenfsdqn
- *   - odtstrenp
- *   - odtstrenn
- *   - adrvstrenp
- *   - adrvstrenn
+ *   - DRVSTRENFSDQP
+ *   - DRVSTRENFSDQN
+ *   - ODTSTRENP
+ *   - ODTSTRENN
+ *   - ADRVSTRENP
+ *   - ADRVSTRENN
  *
  * \return >=0 value on success, else negative.
  */
@@ -35,9 +35,7 @@ int ddrphy_phyinit_mapdrvstren(int drvstren_ohm, enum drvtype targetcsr)
 {
 	int stren_setting = -1;
 
-	if ((targetcsr == drvstrenfsdqp) || (targetcsr == drvstrenfsdqn)) {
-		/* drvstrenfsdqp or drvstrenfsdqn */
-
+	if ((targetcsr == DRVSTRENFSDQP) || (targetcsr == DRVSTRENFSDQN)) {
 		if (drvstren_ohm == 0) {
 			stren_setting = 0x00; /* High-impedance */
 		} else if (drvstren_ohm < 29) {
@@ -77,9 +75,7 @@ int ddrphy_phyinit_mapdrvstren(int drvstren_ohm, enum drvtype targetcsr)
 		} else {
 			stren_setting = 0x00; /* High-impedance */
 		}
-	} else if (targetcsr == odtstrenp) {
-		/* odtstrenp */
-
+	} else if (targetcsr == ODTSTRENP) {
 #if STM32MP_DDR3_TYPE
 		/*
 		 * DDR3 - P and N has the same impedance and non-zero
@@ -170,9 +166,7 @@ int ddrphy_phyinit_mapdrvstren(int drvstren_ohm, enum drvtype targetcsr)
 		/* LPDDR4 - P is high-Z */
 		stren_setting = 0x00; /* High-impedance */
 #endif /* STM32MP_LPDDR4_TYPE */
-	} else if (targetcsr == odtstrenn) {
-		/* odtstrenn */
-
+	} else if (targetcsr == ODTSTRENN) {
 #if STM32MP_DDR3_TYPE
 		/*
 		 * DDR3 - P and N has the same impedance and non-zero
@@ -262,9 +256,7 @@ int ddrphy_phyinit_mapdrvstren(int drvstren_ohm, enum drvtype targetcsr)
 			stren_setting = 0x00; /* High-impedance */
 		}
 #endif /* STM32MP_LPDDR4_TYPE */
-	} else if ((targetcsr == adrvstrenp) || (targetcsr == adrvstrenn)) {
-		/* adrvstrenp or adrvstrenn */
-
+	} else if ((targetcsr == ADRVSTRENP) || (targetcsr == ADRVSTRENN)) {
 		if (drvstren_ohm == 120) {
 			stren_setting = 0x00;
 		} else if (drvstren_ohm == 60) {
diff --git a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_reginterface.c b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_reginterface.c
index 66755cf135..3d10529a0a 100644
--- a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_reginterface.c
+++ b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_reginterface.c
@@ -38,11 +38,11 @@
  * Array of Address/value pairs used to store register values for the purpose
  * of retention restore.
  */
-#define RETREG_AREA	((MAX_NUM_RET_REGS + 1) * sizeof(reg_addr_val_t))
+#define RETREG_AREA	((MAX_NUM_RET_REGS + 1) * sizeof(struct reg_addr_val))
 #define RETREG_BASE	(RETRAM_BASE + RETRAM_SIZE - RETREG_AREA)
 
 static int *retregsize = (int *)(RETREG_BASE);
-static reg_addr_val_t *retreglist = (reg_addr_val_t *)(RETREG_BASE + sizeof(int));
+static struct reg_addr_val *retreglist = (struct reg_addr_val *)(RETREG_BASE + sizeof(int));
 
 static int numregsaved; /* Current Number of registers saved. */
 static int tracken = 1; /* Enabled tracking of registers */
@@ -54,7 +54,7 @@ static int tracken = 1; /* Enabled tracking of registers */
  * during PhyInit registers writes, keeps track of address
  * for the purpose of restoring the PHY register state during PHY
  * retention exit process.  Tracking can be turned on/off via the
- * ddrphy_phyinit_reginterface starttrack, stoptrack instructions. By
+ * ddrphy_phyinit_reginterface STARTTRACK, STOPTRACK instructions. By
  * default tracking is always turned on.
  *
  * \return 0 on success.
@@ -97,7 +97,7 @@ int ddrphy_phyinit_trackreg(uint32_t adr)
  * Register tracking is enabled by calling:
  *
  *  \code
- *  ddrphy_phyinit_reginterface(starttrack,0,0);
+ *  ddrphy_phyinit_reginterface(STARTTRACK,0,0);
  *  \endcode
  *
  * from this point on any call to ddrphy_phyinit_usercustom_io_write16() in
@@ -105,25 +105,25 @@ int ddrphy_phyinit_trackreg(uint32_t adr)
  * ddrphy_phyinit_trackreg(). Tracking is disabled by calling:
  *
  *  \code
- *  ddrphy_phyinit_reginterface(stoptrack,0,0);
+ *  ddrphy_phyinit_reginterface(STOPTRACK,0,0);
  *  \endcode
  *
  * On calling this function, register write via
  * ddrphy_phyinit_usercustom_io_write16 are no longer tracked until a
- * starttrack call is made.  Once all the register write are complete, saveRegs
+ * STARTTRACK call is made.  Once all the register write are complete, SAVEREGS
  * command can be issue to save register values into the internal data array of
- * the register interface.  Upon retention exit restoreregs are command can be
+ * the register interface.  Upon retention exit RESTOREREGS are command can be
  * used to issue register write commands to the PHY based on values stored in
  * the array.
  *  \code
- *   ddrphy_phyinit_reginterface(saveregs,0,0);
- *   ddrphy_phyinit_reginterface(restoreregs,0,0);
+ *   ddrphy_phyinit_reginterface(SAVEREGS,0,0);
+ *   ddrphy_phyinit_reginterface(RESTOREREGS,0,0);
  *  \endcode
  * \return 0 on success.
  */
-int ddrphy_phyinit_reginterface(reginstr myreginstr, uint32_t adr, uint16_t dat)
+int ddrphy_phyinit_reginterface(enum reginstr myreginstr, uint32_t adr, uint16_t dat)
 {
-	if (myreginstr == saveregs) {
+	if (myreginstr == SAVEREGS) {
 		int regindx;
 
 		/*
@@ -141,7 +141,7 @@ int ddrphy_phyinit_reginterface(reginstr myreginstr, uint32_t adr, uint16_t dat)
 		*retregsize = numregsaved;
 
 		return 0;
-	} else if (myreginstr == restoreregs) {
+	} else if (myreginstr == RESTOREREGS) {
 		int regindx;
 
 		/*
@@ -154,16 +154,16 @@ int ddrphy_phyinit_reginterface(reginstr myreginstr, uint32_t adr, uint16_t dat)
 		}
 
 		return 0;
-	} else if (myreginstr == starttrack) { /* Enable tracking */
+	} else if (myreginstr == STARTTRACK) { /* Enable tracking */
 		tracken = 1;
 		return 0;
-	} else if (myreginstr == stoptrack) { /* Disable tracking */
+	} else if (myreginstr == STOPTRACK) { /* Disable tracking */
 		tracken = 0;
 		return 0;
-	} else if (myreginstr == dumpregs) { /* Dump restore state to file. */
+	} else if (myreginstr == DUMPREGS) { /* Dump restore state to file. */
 		/* TBD */
 		return 0;
-	} else if (myreginstr == importregs) { /* import register state from file. */
+	} else if (myreginstr == IMPORTREGS) { /* import register state from file. */
 		/* TBD */
 		return 0;
 	} else {
diff --git a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_restore_sequence.c b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_restore_sequence.c
index e59ab54804..108c8ebe5c 100644
--- a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_restore_sequence.c
+++ b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_restore_sequence.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -58,7 +58,7 @@ int ddrphy_phyinit_restore_sequence(void)
 	mmio_write_16((uintptr_t)(DDRPHYC_BASE + 4 * (TMASTER | CSR_CALZAP_ADDR)), 0x1U);
 
 	/* -# Issue register writes to restore registers values. */
-	ret = ddrphy_phyinit_reginterface(restoreregs, 0, 0);
+	ret = ddrphy_phyinit_reginterface(RESTOREREGS, 0, 0);
 	if (ret != 0) {
 		return ret;
 	}
diff --git a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_sequence.c b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_sequence.c
index 620245c7ac..9774dc3eba 100644
--- a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_sequence.c
+++ b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_sequence.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -58,7 +58,7 @@ int ddrphy_phyinit_sequence(bool skip_training, bool reten)
 	ddrphy_phyinit_usercustom_custompretrain();
 
 	/* Stop retention register tracking for training firmware related registers */
-	ret = ddrphy_phyinit_reginterface(stoptrack, 0, 0);
+	ret = ddrphy_phyinit_reginterface(STOPTRACK, 0, 0);
 	if (ret != 0) {
 		return ret;
 	}
@@ -104,7 +104,7 @@ int ddrphy_phyinit_sequence(bool skip_training, bool reten)
 	}
 
 	/* Start retention register tracking for training firmware related registers */
-	ret = ddrphy_phyinit_reginterface(starttrack, 0, 0);
+	ret = ddrphy_phyinit_reginterface(STARTTRACK, 0, 0);
 	if (ret != 0) {
 		return ret;
 	}
diff --git a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_storeincvfile.c b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_storeincvfile.c
index 8083c5763a..6b93eb6524 100644
--- a/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_storeincvfile.c
+++ b/drivers/st/ddr/phy/phyinit/src/ddrphy_phyinit_storeincvfile.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -21,7 +21,7 @@
  * offset.
  */
 int ddrphy_phyinit_storeincvfile(char *incv_file_name, int mem[],
-				 return_offset_lastaddr_t return_type)
+				 enum return_offset_lastaddr return_type)
 {
 	FILE *incvfile_ptr;
 	char *p;
@@ -64,7 +64,7 @@ int ddrphy_phyinit_storeincvfile(char *incv_file_name, int mem[],
 	}
 	fclose(incvfile_ptr);
 
-	if (return_type == return_lastaddr) {
+	if (return_type == RETURN_LASTADDR) {
 		offset = adr; /*return the last addr */
 	}
 
diff --git a/drivers/st/ddr/phy/phyinit/usercustom/ddrphy_phyinit_usercustom_saveretregs.c b/drivers/st/ddr/phy/phyinit/usercustom/ddrphy_phyinit_usercustom_saveretregs.c
index 17019707ad..df19299cdc 100644
--- a/drivers/st/ddr/phy/phyinit/usercustom/ddrphy_phyinit_usercustom_saveretregs.c
+++ b/drivers/st/ddr/phy/phyinit/usercustom/ddrphy_phyinit_usercustom_saveretregs.c
@@ -390,7 +390,7 @@ int ddrphy_phyinit_usercustom_saveretregs(void)
 	 * --------------------------------------------------------------------------
 	 */
 
-	ret = ddrphy_phyinit_reginterface(saveregs, 0, 0);
+	ret = ddrphy_phyinit_reginterface(SAVEREGS, 0, 0);
 	if (ret != 0) {
 		return ret;
 	}
-- 
2.53.0

